### PR TITLE
P#48327(Task) Make contact selectable

### PR DESCRIPTION
**Pull request by hungnc89**

### CODE REVIEW

**Comment by jayay** [comment]
Thanks! Is it possible to cover more lines with tests?

**Comment by hungnc89** [comment]
Hi @jayay,
I added tests for code.
Thanks

**Comment by jayay** [comment]
@hungnc89 Thank you so much! The test is broken, would you please have a look?

**Comment by hungnc89** [comment]
@jayay oh sorry. I fixed it

**Review by jayay — APPROVED** [review]
Thank you!

File: plugin/Model/FormModelBuilder/FormModelBuilderDBForm.php [hunk 1]
```diff
@@ -364,9 +364,14 @@ public function getDataContactType($module)
                 ->buildFieldsCollection($pFieldLoader);
             $fields = $pFieldCollectionAddressEstate->getFieldsByModule($module);
             $result = [];
+			var_dump($fields['ArtDaten']);
```
Comment (jayay) [inline plugin/Model/FormModelBuilder/FormModelBuilderDBForm.php:367]: This looks like debug output

**Review by jayay — CHANGES_REQUESTED** [review]

**Comment by hungnc89** [comment]
@jayay sorry. I fixed it

**Review by jayay — CHANGES_REQUESTED** [review]
There's still a comment of mine, that's not resolved.

File: plugin/Installer/DatabaseChanges.php [hunk 2]
```diff
@@ -689,14 +692,15 @@ private function deleteCommentFieldApplicantSearchForm()
...
+	public function getQueryAddColumnContactTypePluginFormTable() :string
 	{
 		$prefix = $this->getPrefix();
+		$charsetCollate = $this->getCharsetCollate();
 		$tableName = $prefix . "oo_plugin_forms";
 
-		return $this->_pWPDB->query("ALTER TABLE $tableName 
-				ADD COLUMN `contact_type` varchar(255) NULL DEFAULT NULL 
-				");
+		return "CREATE TABLE $tableName (
```
[the remark this answers is not in the document]
Comment (jayay) [reply plugin/Installer/DatabaseChanges.php:712]: The new column literally needs to be added in the method `getCreateQueryForms`.

**Comment by tung-le-esg** [comment]
> There's still a comment of mine, that's not resolved.

@hoangdoegs Could you please check the comments and resolve the conflict file?

**Review by jayay — APPROVED** [review]

**Comment by michaelarnold-dev** [comment]
@hungnc89 Could you please improve the code coverage?

**Comment by hoangdoegs** [comment]
> @hungnc89 Could you please improve the code coverage?

I added more tests. Now the coverage has been improved

**Review by jayay — APPROVED** [review]

**Comment by michaelarnold-dev** [comment]
Comment of Robert Igelmund Requirement: 

"The user can theoretically set the contact type in the scheduled field in "Form Specific Options" (FO) as well as add it to the "Fields" list (FL) so that it is selectable in the frontend form. This may cause a conflict. This case will probably rarely occur in practice. If there is a conflict, the intended field should win in the FO unless no contact type is selected."

The mail to the broker contains in case of conflict the wrong contact type, namely the one selected by the user in the frontend.
The mail should list the contact type that is also written in the address.

That means the error from #2752693 is not fixed with version 2.21.0-27-g8d09228b. See also screenshot.

**Comment by nglelinh** [comment]
@jayay please help to confirm that the last feedback from Robert is not the WP issue

**Comment by jayay** [comment]
@nglelinh The screenshot appears to be correct, but sadly, is in German. It depicts the field _kind of contact_ in the e-mail to be different than the _kind of contact_ of the newly created address, which is the behavior @michaelarnold-dev is referring to here: https://github.com/onOfficeGmbH/oo-wp-plugin/pull/88#issuecomment-1022180401.

**Comment by hoangdoegs** [comment]
@jayay Please check again

**Review by jayay — APPROVED** [review]

**Review by jayay — APPROVED** [review]

**Pull request merged**